source/main/windows/windowBounds.js

```diff
@@ -0,0 +1,110 @@
+// @flow
+import { debounce } from 'lodash-es/function';
+
+interface Rectangle {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+}
+
+interface Window {
+  getBounds(): Rectangle;
+  on(event: string, handler: Function): void;
+}
+
+interface Display {
+  workArea: Rectangle;
+}
+
+interface Screen {
+  getDisplayMatching(rect: Rectangle): Display;
+  workArea: Rectangle;
+}
+
+interface Store {
+  set: (key: string, value: Rectangle) => void;
+  get: (key: string) => ?Rectangle;
+}
+
+const windowBoundsDefaultStoreKey = 'windowBounds';
+
+function getRightEdge(rect: Rectangle) {
+  return rect.x + rect.width;
+}
+
+function getBottomEdge(rect: Rectangle) {
+  return rect.y + rect.height;
+}
+
+function isWithinBounds(rect: Rectangle, bounds: Rectangle): boolean {
+  return (
+    rect.x >= bounds.x &&
+    rect.y >= bounds.y &&
+    getRightEdge(rect) <= getRightEdge(bounds) &&
+    getBottomEdge(rect) <= getBottomEdge(bounds)
+  );
+}
+
+function isFittingIntoBounds(rect: Rectangle, bounds: Rectangle): boolean {
+  return rect.width <= bounds.width && rect.height <= bounds.height;
+}
+
+function getCenteredRectInBounds(
+  rect: Rectangle,
+  bounds: Rectangle
+): Rectangle {
+  return {
+    width: rect.width,
+    height: rect.height,
+    x: bounds.x + (bounds.width - rect.width) / 2,
+    y: bounds.y + (bounds.height - rect.height) / 2,
+  };
+}
+
+export function saveWindowBounds(
+  window: Window,
+  store: Store,
+  storeKey: string = windowBoundsDefaultStoreKey
+): void {
+  store.set(storeKey, window.getBounds());
+}
+
+/**
+ * Fetches the saved window bounds from provided store and
+ * applies our business rules regarding window management.
+ */
+export function restoreSavedWindowBounds(
+  screen: Screen,
+  store: Store,
+  storeKey: string = windowBoundsDefaultStoreKey
```

Comment:
If the third argument is never passed in the entire code, why do we need to define it as an argument? Could it be it is going to be used in the future?

Reply:
I would say that's just clean code because otherwise, these functions would hardcode the store value which might not be a big issue here but in general i think this is a cleaner, more re-usable approach.